### PR TITLE
Add conjunction and disjunction of authorizers

### PR DESCRIPTION
**Pull request by ihostage**
Moved from Lagom-Pac4j. See https://github.com/pac4j/lagom-pac4j/pull/5

### CODE REVIEW

File: pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/CheckHttpMethodAuthorizer.java [hunk 1]
```diff
@@ -35,4 +35,16 @@ protected boolean check(final WebContext context, final CommonProfile profile, f
         final String requestMethod = context.getRequestMethod();
         return requestMethod.equalsIgnoreCase(element.toString());
     }
+
+    public static CheckHttpMethodAuthorizer requireAllPermissions(HTTP_METHOD... methods) {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/CheckHttpMethodAuthorizer.java]: I think there is a failed copy/paste here. `checkHttpMethod`  is the right method name...

File: pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/AndAuthorizer.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import static java.util.Arrays.asList;
+
+/**
+ * The conjunction of authorizers.
+ * @param <U> Type of profile
+ *
+ * @author Sergey Morgunov
+ * @since 3.4.0
+ */
+public class AndAuthorizer<U extends CommonProfile> implements Authorizer<U> {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/AndAuthorizer.java]: Can you add a test class for this `AndAuthorizer`? My level of test expectations is higher in pac4j-core...

File: pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/OrAuthorizer.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import static java.util.Arrays.asList;
+
+/**
+ * The disjunction of authorizers.
+ * @param <U> Type of profile
+ *
+ * @author Sergey Morgunov
+ * @since 3.4.0
+ */
+public class OrAuthorizer<U extends CommonProfile> implements Authorizer<U> {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/OrAuthorizer.java]: Can you add a test class for this `OrAuthorizer `?

File: pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/CheckHttpMethodAuthorizer.java (hunk 1 above)
Comment (ihostage) [reply pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/CheckHttpMethodAuthorizer.java]: Oops, sorry 😞 
Fixed

File: pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/AndAuthorizer.java (hunk 2 above)
Comment (ihostage) [reply pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/AndAuthorizer.java]: Yes, of course. 
Tests added.

File: pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/OrAuthorizer.java (hunk 3 above)
Comment (ihostage) [reply pac4j-core/src/main/java/org/pac4j/core/authorization/authorizer/OrAuthorizer.java]: Added.

**Comment by leleuj** [comment]
The build fails due to PMD violations. Please have a look.

The documentation needs to be updated as well.

**Comment by ihostage** [comment]
Please, review 😄 
1. I suppress PMD violations because static imports used for increase readability of code in this case.
2. Documentation updated

**Comment by leleuj** [comment]
Excellent! Merging

**Pull request merged**